pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -124,6 +124,23 @@ func MergeConditions(oldConditions []gardencorev1beta1.Condition, newConditions
 	return out
 }
 
+// RemoveConditions removes the conditions with the given types from the given conditions slice.
+func RemoveConditions(conditions []gardencorev1beta1.Condition, conditionTypes ...gardencorev1beta1.ConditionType) []gardencorev1beta1.Condition {
+	conditionTypesMap := make(map[gardencorev1beta1.ConditionType]struct{})
```

Comment:
nit (we already know the required map capacity)
```suggestion
	conditionTypesMap := make(map[gardencorev1beta1.ConditionType]struct{}, len(conditionTypes))
```

Reply:
Done